test/test_component_derived/Nested.HierarchyD.cpp

```diff
@@ -21,10 +21,10 @@ namespace winrt::test_component_derived::Nested::implementation
         test_component_base::HierarchyA a = *this;
         assert(a);
 
-        Nested::IHierarchyD id = *this;
+        Nested::IHierarchyD id = d;
```

Comment:
This is harder to solve, perhaps not that important but I'm not sure how much code depends on this working.